modules/apps/sync/sync-engine/src/main/java/com/liferay/sync/engine/file/system/SyncWatchEventProcessor.java

```diff
@@ -360,6 +361,16 @@ protected void addFolder(SyncWatchEvent syncWatchEvent) throws Exception {
 			FileKeyUtil.writeFileKey(
 				targetFilePath, String.valueOf(syncFile.getSyncFileId()), true);
 		}
+		else if (targetFilePath.toString().equalsIgnoreCase(
+					sourceFilePath.toString())) {
+
+			// Handle edge case where changing filename case will trigger
```

Comment:
I think this only happens on Mac. If I create a folder and change the case immediately after, I get a database exception on the FPN unique index because a create event occurs after a rename event. This handles that case.